makefiles/boards/stm32.inc.mk

```diff
@@ -24,7 +24,7 @@ ifeq (dfu-util,$(PROGRAMMER))
   # with blackmagic_dfu, see https://github.com/blacksphere/blackmagic/
   # To stop bootloader from loading an existing firmware, pull down
   # (ground) GPIO B1.
-  ifeq (,$(DFU_USB_ID))
+  ifeq (,$(DFU_USB_ID) $(filter usbus_dfu,$(USEMODULE)))
```

Comment:
I don't know, I guess this is for `blxxxpill` boards only.

Reply:
We should ask @maribu if this check can be removed, he added the check with commit 327060ca40b08cf7d0cd614f4e0ea3970deb320c.

Reply:
> It's a bit odd that apparently only STM32 boards need this variable set when there are certainly other boards that can come with a DFU bootloader.

If there is only one device in DFU mode it works without specifying a certain device. Even more `dfu-util` accepts the malformed option `--device :` if `USB_VID` and `USB_PID` are not defined.

Reply:
Here we run into a problem anyway with boards that are shipped with a DFU bootloader, like `weact-f4x1cx`, `nz32-sc151`, `patricle-*`, ... or that might use a DFU bootloader, like the `bluepill` board. All these boards use `dfu-util` as default or alternative programmer with a fixed predefined `DFU_USB_ID`. Using `riotboot_dfu` is only possible for these boards if `DFU_USB_ID` is defined in the make command to override the predefined `DFU_USB_ID`.

@dylad For such boards, the definition of `DFU_USB_ID` in
```
FEATURES_REQUIRED+=riotboot USEMODULE+=usbus_dfu make -C examples/saul BOARD=particle-xenon \
PROGRAMMER=dfu-util DFU_USB_ID=1209:7d02 all riotboot/flash-slot0
```
is mandatory to be able to use `riotboot_dfu`.

Therefore, I suggest:
- We remove the check for `DFU_USB_ID` here since all boards that use `dfu-util` as programmer define the `DFU_USB_ID` variable. If not, `dfu-util` simply uses the found DFU device if there is only one of them. @maribu would you be OK with that?
- Either we document that the definition of the variable `DFU_USB_ID` in the make command is mandatory for using `riotboot_dfu`, or we overwrite the board definition of `DFU_USB_ID` in `makefiles/usb-codes.inc.mk`.
   ```diff
   +ifneq (,$(filter usbus_dfu,$(USEMODULE)))
   +  DFU_USB_ID = 1209:7D02
   +endif
   ```
   I would prefere the later one.

Reply:
Sounds reasonable to me

Reply:
BTW, with some changes I will provide in next few days, the `bluepill` could also use `riotboot_dfu` to be flashed via USB.